src/box/errcode.h

```diff
@@ -292,6 +292,7 @@ struct errcode_record {
 	/*237 */_(ER_CREATE_FOREIGN_KEY,	"Failed to create foreign key '%s' in space '%s': %s") \
 	/*238 */_(ER_FOREIGN_KEY_INTEGRITY,	"Foreign key '%s' integrity check failed: %s") \
 	/*239 */_(ER_FIELD_FOREIGN_KEY_FAILED,	"Foreign key constraint '%s' failed for field '%s': %s") \
+	/*239 */_(ER_COMPLEX_FOREIGN_KEY_FAILED, "Foreign key constraint '%s' failed: %s") \
```

Comment:
I didn't review this patch yet, but I have a question: why do you need to separate multipart foreign keys from single part ones? Aren't they essentially the same thing? Wouldn't it be better to squash this and the previous commits?

Reply:
At least they have different error messages. But the most important part is that field foreign keys, like field constraints, a more optimal.